docs/reference/ci-cd.md

```diff
@@ -31,3 +37,24 @@ clusters.
 :::{note}
 TODO: This is yet small, it needs to be extended to talk more about our CI/CD system.
 :::
+
+## Helm chart values and cluster config validation
+
+All of our [helm charts](https://github.com/2i2c-org/infrastructure/tree/HEAD/helm-charts) have associated `values.schema.yaml` files and we also maintain a custom [`cluster.schema.yaml` file](https://github.com/2i2c-org/infrastructure/blob/HEAD/shared/deployer/cluster.schema.yaml).
+Here is an example of a [`values.schema.yaml` file from the basehub chart](https://github.com/2i2c-org/infrastructure/blob/HEAD/helm-charts/basehub/values.schema.yaml).
+These schemas explicitly list what values can be passed through our config, and what type these values should have.
+Therefore, we can use these schemas to validate all our config before making deploys, and catch bugs early.
+
+We have [functions within the deployer](https://github.com/2i2c-org/infrastructure/blob/HEAD/deployer/__main__.py#L213-L302) that validate the cluster config, the support chart values, and the helm chart values for each hub against these schemas.
+We automatically run these functions in GitHub Actions configured by the [`validate-clusters.yaml` workflow file](https://github.com/2i2c-org/infrastructure/blob/master/.github/workflows/validate-clusters.yaml).
+This workflow is triggered when changes to files along the following filepaths are detected.
```

Comment:
I think listing these files isn't sustainable. It breaks the DRY principle without good enough motivation i think.

Whenever we update one we should update the other now or get outdated docs etc. I think if there should be an explicit comment on the level on when the workflow is triggered, it should be generally descriptive to say "only when related configuration has changed" or similar, omitting the exact details.